core/model/src/main/java/org/eclipse/sw360/antenna/model/util/ArtifactLicenseUtils.java

```diff
@@ -54,13 +57,33 @@ public static LicenseInformation getFinalLicenses(Artifact artifact) {
         final Optional<LicenseInformation> declared = artifact.askForGet(DeclaredLicenseInformation.class);
         final Optional<LicenseInformation> observed = artifact.askForGet(ObservedLicenseInformation.class);
         if(declared.isPresent() && observed.isPresent()) {
-            final LicenseStatement effective = new LicenseStatement();
-            effective.setLeftStatement(declared.get());
-            effective.setRightStatement(observed.get());
-            effective.setOp(LicenseOperator.AND);
-            return effective;
-        }
+            List<License> allLicenses = new ArrayList<>(
+                    Stream.of(declared.get().getLicenses(), observed.get().getLicenses())
+                            .flatMap(List::stream)
+                            .collect(Collectors.toSet()));
 
+            return createLicenseStatement(allLicenses);
+        }
         return declared.orElse(observed.orElse(new LicenseStatement()));
     }
+
+    private static LicenseInformation createLicenseStatement(List<License> allLicenses) {
```

Comment:
I thought about the computation, I think the best way is that if there is an OR or WITH in the either the declared or the observed statement, we do nothing, it is better to have the effective license not set instead of set wrong.

Damn it, the longer I think about it the more I think that the problem is not the general computation of the formula, the old algorithm was at least correct, but the usage of the formula. Actually we need two things:
1. A validation rule that triggers whenever there is an OR left in the formula during policy evaluation. This has to be selected by the user
2. When the formula is used for e.g. extracting the licenses for the Disclosure Document, we should ensure that every license is shown only once.

But I have to check with Robert about the issue, this will not solve his problem. Let us first clarify the real need, this is more difficult than expected and we should not come up with wrong results.

Reply:
And one more idea, we should check if the declared license is contained in the observed license, then taking the observed license is the way to go, only if this is not the case, we should combine the two with and, what do you think, Onur?

Reply:
> 1. A validation rule that triggers whenever there is an OR left in the formula during policy evaluation. This has to be selected by the user

This is something for an Independent validator or for the rule engine.

Reply:
Exactly, that was what I meant with policy evaluation -> rule engine

Reply:
Oh sorry, I have missed that essential part.